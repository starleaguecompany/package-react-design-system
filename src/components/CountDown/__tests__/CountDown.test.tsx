import * as React from 'react';
import { render, screen, waitFor } from '@testing-library/react';

import { CountDown } from '../index';

describe('CountDown', () => {
  test('should render correctly with no props', () => {
    render(<CountDown />);

    expect(screen.getByTestId('CountDown')).toMatchSnapshot();
    expect(screen.getByTestId('CountDown')).toHaveClass('container', { exact: true });
  });

  test('should render correctly with attributes', () => {
    render(<CountDown id="test-id" className="test-class" />);

    expect(screen.getByTestId('CountDown')).toMatchSnapshot();
    expect(screen.getByTestId('CountDown')).toHaveClass('test-class');
    expect(screen.getByTestId('CountDown')).toHaveAttribute('id', 'test-id');
  });

  test('should render correctly with from, to props', async () => {
    render(<CountDown from={100} to={987654} />);

    expect(screen.getByTestId('CountDown')).toMatchSnapshot();
    expect(screen.getByText('100')).toBeVisible();
    expect(screen.queryByText('987654')).not.toBeInTheDocument();
    await waitFor(() => {
      expect(screen.getByText('987654')).toBeVisible();
      expect(screen.queryByText('100')).not.toBeInTheDocument();
    });
  });

  test('should render correctly with speed prop', async () => {
    render(<CountDown from={100} to={987654} speed={1000} />);

    expect(screen.getByTestId('CountDown')).toMatchSnapshot();
    await waitFor(() => {
      expect(screen.queryByText('987654')).not.toBeInTheDocument();
    });
    await waitFor(
      () => {
        expect(screen.getByText('987654')).toBeVisible();
      },
      { timeout: 1500 }
    );
  });

  test('should render correctly with interval prop', async () => {
    render(<CountDown from={100} to={987654} interval={500} />);

    expect(screen.getByTestId('CountDown')).toMatchSnapshot();
    await waitFor(
      () => {
        expect(screen.getByText('100')).toBeVisible();
      },
      { timeout: 500 }
    );
    await waitFor(
      () => {
        expect(screen.queryByText('100')).not.toBeInTheDocument();
      },
      { timeout: 501 }
    );
  });

  const formatter = (number: number) => {
    const formatter = new Intl.NumberFormat('ru-RU', {
      style: 'currency',
      currency: 'RUB',
      minimumFractionDigits: 2,
      maximumFractionDigits: 2,
    });

    return formatter.format(+number);
  };

  test('should render correctly with formatter prop', async () => {
    render(<CountDown from={100} to={987654} formatter={formatter} />);

    expect(screen.getByTestId('CountDown')).toMatchSnapshot();
    expect(screen.getByText('100,00 ₽')).toBeVisible();
    await waitFor(() => {
      expect(screen.getByText('987 654,00 ₽')).toBeVisible();
    });
  });
});
