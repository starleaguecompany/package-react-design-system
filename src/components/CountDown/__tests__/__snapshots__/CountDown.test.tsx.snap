// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountDown should render correctly with attributes 1`] = `
<span
  class="test-class container"
  data-qa="CountDown"
  id="test-id"
>
  0
</span>
`;

exports[`CountDown should render correctly with formatter prop 1`] = `
<span
  class="container"
  data-qa="CountDown"
>
  100,00 ₽
</span>
`;

exports[`CountDown should render correctly with from, to props 1`] = `
<span
  class="container"
  data-qa="CountDown"
>
  100
</span>
`;

exports[`CountDown should render correctly with interval prop 1`] = `
<span
  class="container"
  data-qa="CountDown"
>
  100
</span>
`;

exports[`CountDown should render correctly with no props 1`] = `
<span
  class="container"
  data-qa="CountDown"
>
  0
</span>
`;

exports[`CountDown should render correctly with speed prop 1`] = `
<span
  class="container"
  data-qa="CountDown"
>
  100
</span>
`;
